### ISSUE TITLE
expose thing last seen time stamp in UI

### ISSUE THREAD

**Issue by kgiori** [mozilla-iot/gateway#1846]
Especially for battery operated devices (e.g., door and motion sensors, wireless pushbuttons, alarms) it would be nice to know when a device was last "seen". That is, when did the last communication with the device occur? My battery-operated Zigbee devices fail silently when the battery dies, and I don't realize they are offline until I test them. If I could see the last communication timestamp on the thing detail page, it would help me discover which devices have dead batteries. 

Another thought is to have a configurable "last seen" time threshold. It could be a value applied globally, or per add-on, or even per device. The setting would let a user set a threshold time period after which the gateway could (optionally) be configured to notify them that a device is offline. For example, if not seen within X seconds, show device as "disconnected" and optionally trigger a "<DEVICE> last seen <TIMESTAMP>" alert or something like that. Maybe the last seen timestamp data could always be shown as a subtitle on a Thing's "Events" page. The page would then be more interesting for devices that don't produce events.

**Comment by raidnet-ms** [comment]
A notification alert for "low battery" in the schema would be useful too.
A watchdog server could check for low battery and could be used by clients to restart themself if no watchdog server is found.

**Comment by mrstegeman** [comment]
@raidnet-ms See #1439

**Comment by benfrancis** [comment]
If the use case is that you want to be made aware when a device's battery has died, then I agree with @raidnet-ms that low battery notifications are a good solution. That way you are given an opportunity to take action before it dies.

**Comment by flatsiedatsie** [comment]
In my experience last seen is very useful for moment when you wonder "is this device even working/communicating?"

But you don't need to see it every day. My suggestion: add it to the [map of devices](https://github.com/mozilla-iot/gateway/issues/1842).

**Comment by kgiori** [comment]
I like the idea of a device list with unique network or ID information as well as a last seen timestamp. But I would separate the device list by adapter types -- the sub-lists could be shown on a new adapter-specific page, that is loaded by clicking an adapter name on the `.../settings/adapters` page.

My IP devices (Wi-Fi, Ethernet) would have IP addresses (and/or mdns names?) and show up on the ThingURLAdapter page. My Zigbee devices have a unique ID and would show up on the ZigbeeAdapter page. And so on...

In other words, some of the adapters listed on the settings/adapters page would be clickable and take you to a new web page that shows a device list with last seen timestamp. This per adapter page might also be the best place to "unpair" devices that need or can invoke active unpairing (rather than complicate the Things page). It might also be a good place for per adapter detailed debugging insights, such as in-progress discovery of devices that are not yet fully ready for use.
